### PR TITLE
[nightly-test] add 4-nodes shuffle-data-loader test

### PR DESCRIPTION
**Pull request by scv119**
add shuffle_data_loader_4_nodes that run shuffle across 4 nodes with 400million rows.
Test plan:
-[x] schedule a run https://beta.anyscale.com/o/anyscale-internal/projects/prj_SVFGM5yBqK6DHCfLtRMryXHM/clusters/ses_Fie17wKrUbaLTRFKuDtQa6ee

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]

**Comment by scv119** [comment]
Tests passing: https://beta.anyscale.com/o/anyscale-internal/projects/prj_SVFGM5yBqK6DHCfLtRMryXHM/clusters/ses_niUBF9vJe5KNn4GNdy8EYU3r?job-section=jobs

**Comment by scv119** [comment]
it depends on https://github.com/ray-project/ray/pull/17187/files

File: release/nightly_tests/shuffle_data_loader/shuffle_data_loader_compute_4_nodes.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+cloud_id: cld_17WvYIBBkdgLwEUNcLeRAE
+region: us-west-2
+
+max_workers: 4
+
+aws:
+    IamInstanceProfile: {"Name": "ray-autoscaler-v1"}
```
Comment (scv119) [inline release/nightly_tests/shuffle_data_loader/shuffle_data_loader_compute_4_nodes.yaml:7]: cc @wuisawesome this fixed worker node permission issue for me.

**Comment by rkooo567** [comment]
Any test failure should be unrelated because it only changes tests.

**Pull request merged**